sources/flight-sql/src/executor/mod.rs

```diff
@@ -1,34 +1,118 @@
-use arrow::error::ArrowError;
-use arrow_flight::{sql::client::FlightSqlServiceClient, Ticket};
+use arrow::{datatypes::Schema, error::ArrowError};
+use arrow_flight::{sql::client::FlightSqlServiceClient, FlightInfo};
 use async_trait::async_trait;
 use datafusion::{
     error::{DataFusionError, Result},
     physical_plan::{stream::RecordBatchStreamAdapter, SendableRecordBatchStream},
 };
 use datafusion_federation_sql::SQLExecutor;
-use futures::{executor, StreamExt};
+use futures::{executor, TryStreamExt};
+
 use std::sync::Arc;
-use tonic::transport::Endpoint;
+use tonic::transport::{Certificate, ClientTlsConfig, Endpoint, Identity};
+
+/// Authentication options for FlightSQL Endpoints
+#[derive(Clone)]
+pub enum FlightSQLAuth {
+    Basic(BasicAuth),
+    PKI(PKIAuth),
+    Unsecured,
+}
+
+/// Basic username and password authorization for FlightSQL Endpoints
+#[derive(Clone)]
+pub struct BasicAuth {
+    username: String,
+    password: String,
+}
+
+/// Contains options for completing mTLS/PKI authentication configuration of Tonic client
+#[derive(Clone)]
+pub struct PKIAuth {
+    /// The public x509 cert pem encoded used to auth with the FlightSQL server
+    pub client_cert_file: Arc<[u8]>,
+    /// The private key pem encoded used to validate the public client cert
+    pub client_key_file: Arc<[u8]>,
+    /// The bundle of trusted CAcerts for validating the FlightSQL server
+    pub ca_cert_bundle: Arc<[u8]>,
+}
 
 pub struct FlightSQLExecutor {
     context: String,
+    auth: FlightSQLAuth,
 }
 
 impl FlightSQLExecutor {
-    pub fn new(dsn: String) -> Self {
-        Self { context: dsn }
+    pub fn new(dns: String, auth: FlightSQLAuth) -> Self {
```

Comment:
I wasn't quite sure what the name `dsn` meant... did you mean `dns` i.e. the domain name of the flightSQL endpoint?

Reply:
I guess it's an abbreviation for `Data Source Name`.